packages/components/inputs/checkbox-input/src/checkbox.tsx

```diff
@@ -1,33 +1,53 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { ChangeEventHandler } from 'react';
 import styled from '@emotion/styled';
 import { customProperties as vars } from '@commercetools-uikit/design-system';
 import { accessibleHiddenInputStyles } from '@commercetools-uikit/input-utils';
+import {
+  filterAriaAttributes,
+  filterDataAttributes,
+} from '@commercetools-uikit/utils';
+import type { TCheckboxProps } from './checkbox-input';
 
-// accessible input :)
 const Input = styled.input`
   &:focus + div > svg [id$='borderAndContent'] > [id$='border'] {
     stroke: ${vars.borderColorForInputWhenFocused};
   }
 `;
 
-const Checkbox = (props) => {
-  const ref = React.useRef();
+type TInputRef = {
+  indeterminate: boolean;
+};
+
+type TProps = Omit<TCheckboxProps, 'children'> & {
```

Comment:
I think you need to omit `hasError` and `isHovered` props as well as they are not part of the old PropTypes.